components/Modal.tsx

```diff
@@ -3,29 +3,38 @@ import Typography from '@mui/material/Typography';
 import MUIModal from '@mui/material/Modal';
 import { IconButton } from '@mui/material';
 import CloseIcon from '@mui/icons-material/Close';
+import classnames from 'classnames';
+import { useMemo } from 'react';
 
 interface IModalProps {
   isOpen: boolean,
   close: () => void,
-  title: string,
-  children: JSX.Element
+  title?: string,
+  children: JSX.Element,
+  isConstrained?: boolean
 }
 
-export const Modal = ({ isOpen, close, title, children }: IModalProps) => {
+export const Modal = ({ isOpen, close, title, children, isConstrained = true }: IModalProps) => {
+  
+  const modalTitle = useMemo(() => {
+    if (title) return (
+      <Typography className="modal-content-title" variant="h5" component="h3">
+        {title}
+      </Typography>);
+  }, [title]);
+
   return (
     <MUIModal
       open={isOpen}
       onClose={close}
       className="modal-container"
     >
-      <Box className="modal-content">
-        <Typography className="modal-content-title" variant="h5" component="h3">
-          {title} 
-          <IconButton disableFocusRipple={true} className="modal-content-title__icon-button" onClick={close}>
-            <CloseIcon />
-          </IconButton>
-        </Typography>
-        <Typography component="div" id="modal-modal-description" sx={{ mt: 2 }}>
+      <Box className={classnames('modal-content', { 'modal-content--constrained': isConstrained }, { 'modal-content__item-card': !isConstrained })}>
+        <IconButton disableFocusRipple={true} className="modal-content-title__icon-button" onClick={close}>
+          <CloseIcon />
+        </IconButton>
+        {modalTitle}
```

Comment:
What is the benefit of extracting this title into a separate component (or sub-component) in comparison with regular conditional rendering like 
```typescript
{title 
? 
 (
<Typography className="modal-content-title" variant="h5" component="h3">
    {title}
</Typography> 
)
: null }
```

And do we need to hide the entire component, or it is just enough to hide an empty text?
``` typescript
  {title || ''}
```

Reply:
Yep, you correct as usual. Changed the code to what you suggested.